api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1721,75 +1722,74 @@ type AzurePlatformSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location is immutable"
 	// +immutable
+	// +required
 	Location string `json:"location"`
 
 	// ResourceGroupName is the name of an existing resource group where all cloud resources created by the Hosted
-	// Cluster are to be placed.
+	// Cluster are to be placed. The resource group is expected to exist under the same subscription as SubscriptionID.
+	//
 	// In ARO HCP, this will be the managed resource group where customer cloud resources will be created.
 	//
-	// Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
+	// Resource group naming requirements can be found here: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ResourceGroup.Name/.
+	//
+	//Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
 	//          ResourceGroupName is <resourceGroupName>.
 	//
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
 	// +immutable
+	// +required
 	ResourceGroupName string `json:"resourceGroup"`
 
-	// VnetName is the resource name of an existing VNET to use in creating VMs. If this field is included, it should
-	// be the resource name matching the VnetID.
-	// In ARO HCP, this will be the name of the customer provided VNET.
+	// VnetID is the ID of an existing VNET to use in creating VMs. This VNET is expected to exist under the same
+	// subscription as SubscriptionID.
 	//
-	// Example: if your VNET ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>,
-	//          your VnetName is <vnetName>
-	//
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetName is immutable"
-	// +immutable
-	VnetName string `json:"vnetName"`
-
-	// VnetID is the ID of an existing VNET to use in creating VMs.
 	// In ARO HCP, this will be the ID of the customer provided VNET.
 	//
 	// Example: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>
 	//
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetID is immutable"
-	// +optional
 	// +immutable
+	// +required
 	VnetID string `json:"vnetID,omitempty"`
```

Comment:
I understand that `VnetId` can be in the same, or in a different resource group name than the `ResourceGroupName` attribute right?

If that's correct, we should document so.

Reply:
Will update.